core/src/main/java/tech/pegasys/web3signer/core/multikey/metadata/AzureSecretSigningMetadata.java

```diff
@@ -62,8 +73,36 @@ public String getSecretName() {
     return secretName;
   }
 
+  public AzureAuthenticationMode getAuthenticationMode() {
+    return authenticationMode;
+  }
+
   @Override
   public ArtifactSigner createSigner(final ArtifactSignerFactory factory) {
     return factory.create(this);
   }
+
+  @Override
+  public void validate() throws SigningMetadataException {
```

Comment:
nit: I can't put my finger on it (and it could be nothing) - but having a validate method on this class, implies that this class knows what's valid or not - but doesn't know _how_ its being used, so not sure how this logic can exist ... but equally sure I've argued in the opposite direction before.